docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+forward connections onto the zero-trust network.
+There is some security risk in doing this,
+as the OpenZiti Tunneler can't tell the difference between
+a non-zitified client and malware running on the host.
+
+The green boxes,
+showing Vault, Redis, and Consul,
+are existing third-party non-zitified components
+that are already used by EdgeX.
+These components have independent authentication mechanisms.
+They are safe to use for kernel-mediated same-host access,
+but not safe to access over the network without
+additional network security.
+The OpenZiti Router also has an outbound tunneler built-in,
+allowing traffic to traverse the zero-trust network encrypted,
+and exit unencrypted to communicate with services
+listening on the docker network (zero-trust network access, ZTNA),
+or listening on the host network (zero-trust host access, ZTHA).
+These connections are indicated by the blue-violet arrows.
+Note that Vault also has a thick dashed border,
+indicating that it also must have an exposed listening port.
+This is because a Vault-based identity claim is required
+to join the zero-trust network in the normal case,
+and thus Vault must be available as a prerequisite
+to the zero-trust network being available.
+In EdgeX 3.0,
+Vault is exposed behind a TLS-enabled NGINX API gateway (not shown),
+providing confidentiality and integrity protection for those connections.
+
+Connections to exposed ports on the network underlay
+are denoted with red arrows:
+they always terminate at the boxes with thick dashed borders.
+Vault is special in that it also has an alias
+on the zero-trust network to service internal requests
+to the key-value secret store and the consul secrets engine.
+
+The gray boxes in the "zero-trust network" box are
+the Zitified EdgeX services.
+One might assume that connections between these services are peer-to-peer,
+but this is not the case.
+Instead, EdgeX services connect to the OpenZiti edge router
+which perform packet routing functions.
+The edge router applies network policies to the traffic,
+controlling who can speak to whom,
+but payloads are encrypted end-to-end.
+Only the receiving microservice can decrypt the payload,
+and the receiver has cryptographic proof of the sending service's identity.
+
+### Feature overlaps with EdgeX JWT-based authentication
+
+OpenZiti focuses on being a transport-level solution,
+controlling access to remove services.
+It does not provide application-level security.
+The use of an OpenZiti tunneler causes remote connections
+appear to come from the tunneler instead of the true client.
+In this case,
+use of JWT authentication could be used to authenticate the true client;
+for example, a `curl` client.
+
+Another case where this may apply is if an adopter elects to keep
+an NGINX API gateway as part of the solution.
+Since NGINX intercepts all requests to the backend,
+and does not itself integrate directly with OpenZiti,
+NGINX would need to use an OpenZiti tunneler to forward
```

Comment:
I'd think they could also choose to use this nginx module https://github.com/openziti/ngx_http_ziti_module

Reply:
Updates made. Please re-read these two paragraphs.